server/bleep/src/webserver/answer.rs

```diff
@@ -332,6 +345,103 @@ pub(super) async fn _handle(
     Ok(Sse::new(Box::pin(stream)))
 }
 
+#[derive(serde::Deserialize)]
+pub struct Explain {
+    pub relative_path: String,
+    pub line_start: usize,
+    pub line_end: usize,
+    pub branch: Option<String>,
+    pub repo_ref: RepoRef,
+    #[serde(default = "default_thread_id")]
+    pub thread_id: uuid::Uuid,
+}
+
+pub async fn explain(
+    Query(params): Query<Explain>,
+    Extension(app): Extension<Application>,
+    Extension(user): Extension<User>,
+) -> super::Result<impl IntoResponse> {
+    let query_id = uuid::Uuid::new_v4();
+
+    // We synthesize a virtual `/answer` request.
+    let virtual_req = Answer {
+        q: format!(
+            "Explain lines {} - {} in {}",
+            params.line_start, params.line_end, params.relative_path
+        ),
+        repo_ref: params.repo_ref,
+        thread_id: params.thread_id,
+        parent_exchange_id: None,
+    };
+
+    let conversation_id = ConversationId {
+        thread_id: params.thread_id,
+        user_id: user
+            .login()
+            .ok_or_else(|| super::Error::user("didn't have user ID"))?
+            .to_string(),
+    };
+
+    let mut query = parser::parse_nl(&virtual_req.q)
+        .context("failed to parse virtual answer query")?
+        .into_semantic()
+        // We synthesize the query, this should never fail.
+        .unwrap()
+        .into_owned();
+
+    if let Some(branch) = params.branch {
+        query
+            .branch
+            .insert(Literal::Plain(std::borrow::Cow::Owned(branch)));
+    }
+
+    let file_content = app
+        .indexes
+        .file
+        .by_path(&virtual_req.repo_ref, &params.relative_path, None)
+        .await
+        .context("failed to conduct path search")?
```

Comment:
These error messages would confuse me. I'd think we'd explicitly done a path search. Can you rephrase so more clearly conveys that we've tried and failed to fetch a file's content?